sdks/rust/src/sdk.rs

```diff
@@ -12,349 +12,211 @@
+            "http://localhost:{}",
+            port.unwrap_or_else(|| {
+                env::var("AGONES_SDK_GRPC_PORT")
+                    .ok()
+                    .and_then(|s| s.parse().ok())
+                    .unwrap_or(9357)
+            })
+        )
+        .parse()?;
+
+        let builder = tonic::transport::channel::Channel::builder(addr)
+            .keep_alive_timeout(keep_alive.unwrap_or_else(|| Duration::from_secs(30)));
+
+        // let mut root_store = rustls::RootCertStore::empty();
+        // root_store.add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
+        // let mut rusttls_config = rustls::ClientConfig::new();
+        // rusttls_config.alpn_protocols = vec![b"h2".to_vec(), b"http/1.1".to_vec()];
+        // rusttls_config.root_store = root_store;
+        // let tls_config =
+        //     tonic::transport::ClientTlsConfig::new().rustls_client_config(rusttls_config);
+        // builder = builder.tls_config(tls_config)?;
+
+        let channel = builder.connect().await?;
+        let mut client = SdkClient::new(channel.clone());
+        let alpha = Alpha::new(channel);
+
+        // Loop until we connect. The original implementation just looped once
+        // every second up to a maximum of 30 seconds, but it's better for the
+        // external caller to wrap this in their own timeout
+        let mut connect_interval = tokio::time::interval(Duration::from_millis(100));
 
-        // Unfortunately there isn't a native way to block until connected
-        // so we had to roll our own.
-        let mut counter = 0;
         loop {
-            counter += 1;
-            match cli.get_game_server(&req) {
-                Ok(_) => break,
-                Err(e) => {
-                    if counter > 30 {
-                        return Err(ErrorKind::Grpc(e).into());
-                    }
-                    sleep(Duration::from_secs(1));
-                    continue;
-                }
+            connect_interval.tick().await;
+
+            if client.get_game_server(empty()).await.is_ok() {
+                break;
             }
         }
 
-        // Keep both sender and receiver as RPC is canceled when sender or receiver is dropped
-        let (sender, receiver) = cli.health()?;
-        Ok(Sdk {
-            client: Arc::new(cli),
-            health: Arc::new(Mutex::new(Some(sender))),
-            health_receiver: Arc::new(Mutex::new(Some(receiver))),
-            alpha: Arc::new(alpha),
+        Ok(Self {
+            client,
+            alpha,
         })
     }
 
     /// Alpha returns the Alpha SDK
-    pub fn alpha(&self) -> &Arc<Alpha> {
+    #[inline]
+    pub fn alpha(&self) -> &Alpha {
         &self.alpha
     }
 
     /// Marks the Game Server as ready to receive connections
-    pub fn ready(&self) -> Result<()> {
-        let req = sdk::Empty::default_instance();
-        let res = self.client.ready(req).map(|_| ())?;
-        Ok(res)
+    #[inline]
```

Comment:
Curious - what does the `#[inline]` do here?

Reply:
This is actually a good point, I just annotated the function since they are just thin wrappers around the generated tonic code, but I haven't actually looked at how this impacts the callsite due to async, so I'm going to remove all of them for now, easier to add back later if it does indeed help.